internal/service/dns/record_cname_resource_test.go

```diff
@@ -0,0 +1,619 @@
+	var resourceName = "nios_dns_record_cname.test_forbid_reclamation"
+	var v dns.RecordCname
+	canonical := acctest.RandomNameWithPrefix("test-cname") + ".example.com"
+	name := acctest.RandomNameWithPrefix("test-cname") + ".example.com"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccRecordCnameForbidReclamation(canonical, name, "default", "true"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRecordCnameExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "forbid_reclamation", "true"),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccRecordCnameForbidReclamation(canonical, name, "default", "false"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRecordCnameExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "forbid_reclamation", "false"),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccRecordCnameResource_Name(t *testing.T) {
+	var resourceName = "nios_dns_record_cname.test_name"
+	var v dns.RecordCname
+	name1 := acctest.RandomName() + ".example.com"
+	name2 := acctest.RandomName() + ".example.com"
+	canonical := acctest.RandomNameWithPrefix("test-cname") + ".example.com"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccRecordCnameName(canonical, name1, "default"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRecordCnameExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "name", name1),
+				),
+			},
+			// Update and Read
+			{
+				Config: testAccRecordCnameName(canonical, name2, "default"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckRecordCnameExists(context.Background(), resourceName, &v),
+					resource.TestCheckResourceAttr(resourceName, "name", name2),
+				),
+			},
+			// Delete testing automatically occurs in TestCase
+		},
+	})
+}
+
+func TestAccRecordCnameResource_Ttl(t *testing.T) {
+	var resourceName = "nios_dns_record_cname.test_ttl"
+	var v dns.RecordCname
+	canonical := acctest.RandomNameWithPrefix("test-cname") + ".example.com"
+	name := acctest.RandomNameWithPrefix("test-cname") + ".example.com"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(t) },
+		ProtoV6ProviderFactories: acctest.ProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			// Create and Read
+			{
+				Config: testAccRecordCnameTtl(canonical, name, "default", 1000, "true"),
```

Comment:
```suggestion
				Config: testAccRecordCnameTtl(canonical, name, "default", 1000, true),
```
Nitpick : I think this can be passed as bool value rather than a string

Reply:
Done